CMakeLists.txt

```diff
@@ -12,8 +24,25 @@ option(BUILD_TESTS "${PROJECT_NAME} - Build tests" ON)
 option(WITH_PROFILING "${PROJECT_NAME} - Build with profiling flags" OFF)
 option(WITH_C_BINDINGS "Build iron C bindings" ON)
 option(WITH_Python_BINDINGS "Build iron Python bindings" ON)
+CMAKE_DEPENDENT_OPTION(WITH_DIAGNOSTICS "Build iron with diagnostics support" ON
+                       "IS_DEBUG_BUILD" OFF)
+option(PRINT_CONFIG_SUMMARY "Show a summary of the configuration." TRUE)
 set(PACKAGE_CONFIG_DIR "cmake" CACHE STRING "Directory for package config files (relative to CMAKE_INSTALL_PREFIX)")
 
+string(TOUPPER "${REAL_PRECISION}" _uppercase_REAL_PRECISION)
+if(_uppercase_REAL_PRECISION STREQUAL "SINGLE")
+  set(WORKING_REAL_PRECISION_SINGLE)
```

Comment:
This won't have the desired effect, this essentially unsets WORKING_REAL_PRECISION_SINGLE.  In a conditional statement it will evaluate to False.  It needs to set to a value that will evaluate to True in a conditional statement as that is how it is used later.